crates/layout/src/layout.rs

```diff
@@ -414,147 +395,34 @@ impl LayoutManager {
-    layout_id: i32,
-    parent_layout_id: i32,
-    child_index: i32,
-    style: ViewStyle,
-    name: String,
-    measure_func: impl Send + Sync + 'static + Fn(i32, f32, f32, f32, f32) -> (f32, f32),
-) {
-    info!("add_view_measure layoutId {}", layout_id);
-    let layout_measure_func =
-        move |size: Size<Option<f32>>, available_size: Size<AvailableSpace>| -> Size<f32> {
-            let width = if let Some(w) = size.width { w } else { 0.0 };
-            let height = if let Some(h) = size.height { h } else { 0.0 };
-            let available_width = match available_size.width {
-                AvailableSpace::Definite(w) => w,
-                AvailableSpace::MaxContent => f32::MAX,
-                AvailableSpace::MinContent => 0.0,
-            };
-            let available_height = match available_size.height {
-                AvailableSpace::Definite(h) => h,
-                AvailableSpace::MaxContent => f32::MAX,
-                AvailableSpace::MinContent => 0.0,
-            };
-            let result = measure_func(layout_id, width, height, available_width, available_height);
-            Size { width: result.0, height: result.1 }
-        };
-
-    manager().add_style(
-        layout_id,
-        parent_layout_id,
-        child_index,
-        style,
-        name,
-        Some(taffy::node::MeasureFunc::Boxed(Box::new(layout_measure_func))),
-        None,
-        None,
-    )
-}
 
-pub fn remove_view(
-    layout_id: i32,
-    root_layout_id: i32,
-    compute_layout: bool,
-) -> LayoutChangedResponse {
-    manager().remove_view(layout_id, root_layout_id, compute_layout)
-}
-
-pub fn clear_views() {
-    manager().clear()
-}
-
-pub fn set_node_size(
-    layout_id: i32,
-    root_layout_id: i32,
-    width: u32,
-    height: u32,
-) -> LayoutChangedResponse {
-    manager().set_node_size(layout_id, root_layout_id, width, height)
-}
-
-pub fn compute_node_layout(layout_id: i32) -> LayoutChangedResponse {
-    manager().compute_node_layout(layout_id)
-}
-
-pub fn print_layout(layout_id: i32) {
-    let manager = &mut LAYOUT_MANAGER.lock().unwrap();
-    let taffy = &mut TAFFY.lock().unwrap();
-    print_layout_recurse(layout_id, manager, taffy, "".to_string());
-}
+    pub fn print_layout(self, layout_id: i32) {
+        self.print_layout_recurse(layout_id, "".to_string());
+    }
 
-fn print_layout_recurse(layout_id: i32, manager: &LayoutManager, taffy: &Taffy, space: String) {
-    let layout_node = manager.layout_id_to_taffy_node.get(&layout_id);
-    if let Some(layout_node) = layout_node {
-        let layout = taffy.layout(*layout_node);
-        if let Ok(layout) = layout {
-            let name_result = manager.layout_id_to_name.get(&layout_id);
-            if let Some(name) = name_result {
-                println!("{}Node {} {}:", space, name, layout_id);
-                println!("  {}{:?}", space, layout);
-            }
+    fn print_layout_recurse(&self, layout_id: i32, space: String) {
```

Comment:
I'd say either remove the println()s or replace them with log.verbose calls.

Reply:
Apparently our logging crate doesn't like to print to stdout. I only use this for fetch_layout, a binary I use for testing. I made this take a print function instead, and fetch_layout uses println because I want it in stdout.